Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/IndirectILLEnergyTransfer.py

```diff
@@ -262,28 +263,30 @@ def _mask(self, ws, xstart, xend):
 
     def _convert_to_energy(self, ws):
         """
-        Converts the x-axis from raw channel number to energy transfer
+        Converts the x-axis from raw channel number to energy transfer for Doppler mode
         @param ws :: input workspace name
         """
+        from scipy.constants import physical_constants
+        c = physical_constants['speed of light in vacuum'][0]
+        nm = physical_constants['neutron mass energy equivalent in MeV'][0]
 
-        x = mtd[ws].readX(0)
-        size = mtd[ws].blocksize()
-        mid = (x[-1] + x[0])/ 2.
-        scale = 0.001  # from micro ev to mili ev
-
-        factor = size / (size - 1)
-
-        # minus sign is needed
+        bsize = mtd[ws].blocksize()
         if self._doppler_energy != 0:
-            formula = '-(x/{0} - 1)*{1}'.format(mid, self._doppler_energy * scale * factor)
+            # the doppler channels are linear in velocity (not time, neither deltaE)
```

Comment:
Formulae look fine. Would using [`f-strings`](https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python) here improve the readabilty of the formulae?

Reply:
Didn't know about this, thanks for pointing. However in this particular case I'd actually prefer as is; I find it more readable.